sw/device/examples/hello_usbdev/hello_usbdev.c

```diff
@@ -113,7 +132,13 @@ int main(int argc, char **argv) {
   // Call `usbdev_init` here so that DPI will not start until the
   // simulation has finished all of the printing, which takes a while
   // if `--trace` was passed in.
-  usbdev_init(&usbdev);
+  uint32_t gpio_state;
+  CHECK(dif_gpio_all_read(&gpio, &gpio_state) == kDifGpioOk);
+  LOG_INFO("PHY settings: pinflip=%d differential=%d\n",
+	   gpio_state & kPinflipMask, gpio_state & kDiffMask);
+  
+  usbdev_init(&usbdev, gpio_state & kPinflipMask, gpio_state & kDiffMask,
```

Comment:
Note that `usbdev_init` accepts `bool`, which should be 0 or 1, but these AND operations may produce other values. You should remember to write `(gpio & kPinflipMask) != 0` to avoid this.

Reply:
Shouldn't the compiler do this when it does the casy from uint32 to bool?
In any event I will move to setting in variables to avoid the repeated enumerations.

Reply:
Note the LOG output of the new code confirms that the new bool differential is 1 (not kDiffMask==2)

Reply:
We've tried to avoid implicit int to bool casts in other code so far, and instead favour explicitly writing out the comparison that is intended, but I realise this has not made it to the style guide. I realise you don't get much time for contributions, so I'm happy for this kind of fix to happen in a follow-up, rather than blocking this PR.

Reply:
Ok, happy to adjust the style here if I need more changes, otherwise will do separately.

I did change the code to put these in bool variables but am still using the implicit cast.

Reply:
Done